src/Symfony/Component/Validator/Tests/Constraints/TimestampValidatorTest.php

```diff
@@ -0,0 +1,188 @@
+<?php
+
+namespace Symfony\Component\Validator\Tests\Constraints;
+
+use Symfony\Component\Validator\Constraints\Timestamp;
+use Symfony\Component\Validator\Constraints\TimestampValidator;
+use Symfony\Component\Validator\ConstraintValidatorInterface;
+use Symfony\Component\Validator\Test\ConstraintValidatorTestCase;
+
+class TimestampValidatorTest extends ConstraintValidatorTestCase
+{
+
+    protected function createValidator(): ConstraintValidatorInterface
+    {
+        return new TimestampValidator();
+    }
+
+    public function testNullIsValid()
+    {
+        $this->validator->validate(null, new Timestamp());
+
+        $this->assertNoViolation();
+    }
+
+    public function testNegativeNumberIsInvalid()
+    {
+        $this->validator->validate(-1, new Timestamp([
+            'message' => 'myMessage',
+        ]));
+
+        $this->buildViolation('myMessage')
+            ->setParameter('{{ value }}', -1)
+            ->setInvalidValue(-1)
+            ->setCode(Timestamp::INVALID_TIMESTAMP_ERROR)
+            ->assertRaised();
+    }
+
+    public function getGreaterThan()
+    {
+        return [
+            [1719610882, '@1718210882', "Jun 12, 2024, 4:48\u{202F}PM"],
+        ];
+    }
```

Comment:
* Data providers have to be `static`.
* A data provider with only a single record is pointless.
* You only test with absolute dates. We should also test relative date strings (like `'two months ago'`) because that's likely how your constraint is used.
* The date string depends on the ICU version that PHP is compiled against. Have a look at `IcuCompatibilityTrait` which we already use in other tests.